.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+name: CI
+
+on:
+    push:
+        branches: [master, actions]
+    pull_request:
+        branches: [master, actions]
+
+jobs:
+    Setup:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                python-version: [2.7, 3.6, 3.7, 3.8]
+
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v2
+              with:
+                python-version: ${{ matrix.python-version }}
+
+            - name: Install system dependencies
+              run: |
+                sudo apt-get update -y
+                sudo apt install libcurl4-openssl-dev libssl-dev
+                sudo apt-get install libgnutls28-dev
+
+            - name: Install Python dependencies
+              run: |
+                pip install --upgrade pip setuptools py
+                pip install twine wheel coveralls
+                pip install -e .[all]
+
+            - name: Run Black on Python
+              if: matrix.python-version != 2.7
+              run: black --check .
+
+            - name: Run Python manifest completeness
+              run: check-manifest --ignore ".travis-*"
+
+            - name: Run Sphinx documentation
+              run: sphinx-build -qnNW docs docs/_build/html
+
+            - name: Run Pytest
+              run: python setup.py test
+
+            - name: Run Sphinx documentation with doctests
+              run: sphinx-build -qnNW -b doctest docs docs/_build/doctest
```

Comment:
One thing I'm thinking about is how do we keep `run-tests.sh` and GA in sync? With Travis, `.travis.yml` was simply running `run-tests.sh`, so all we had to do was edit `run-tests.sh`. With GA, we would have to edit test commands in two places... Not a big deal, but could get out of sync.

We could perhaps always use `run-tests.sh` and decorate it with various options such as:

```console
$ run-tests.sh --check-black
$ run-tests.sh --check-manifest
$ run-tests.sh --check-sphinx  
...
```

for each independent command.

In this way all tests are centralised in `run-tests.sh`, they can be run locally, or on GA, or on Travis CI.

WDYT?

Reply:
(We can muse IRL also WRT REANA.)

Reply:
Sounds good to me and will be good for reana as well. We can discuss IRL as well. 👍